onboarding/daml/Onboarding.daml

```diff
@@ -58,14 +58,48 @@ data GuiBackendConfigEntry = GuiBackendConfigEntry with
   deriving (Eq, Show)
 
 
-onboarding : [PartyInfo] -> Script Output
+onboarding : [PartyInfo] -> Script BackendConfig
 onboarding partyInfo = do
   scheduler <- allocatePartyOn "Scheduler" (ParticipantName "schedulerParticipant")
   assembler <- allocatePartyOn "Assembler" (ParticipantName "assemblerParticipant")
   parties <- setupParties partyInfo
   mapA_ (setupBic  scheduler assembler parties) partyInfo
-  entries <- mapA (setupBalances parties) partyInfo
-  pure $ Output { guiBackendConfig = entries, parties = map damlPartyId entries }
+  entries <- mapA (setupBank parties) partyInfo
+  pure $ BackendConfig { 
+      assembler = assembler
+    , scheduler = scheduler
+    , banks = entries
+    , parties = map damlPartyId entries 
+    }
+
+extractConfig : [PartyInfo] -> Script [BankEntry]
+extractConfig partyInfo = do
+  let participantNames = fmap (\info -> ParticipantName info.participantName) partyInfo
+  knownParties <- fmap (.party) <$> listKnownParties
+  knownParties <- foldlA
+    (\result participant -> ((++ result) . fmap (.party)) <$> listKnownPartiesOn participant)
+    knownParties
+    participantNames
+  bics <- foldlA
+    (\result party -> do
+      partyBics <- fmap snd <$> queryFilter @BankBIC party (\x -> signatory x == [party])
+      let partyBicMapping = TextMap.fromList $ fmap (\x -> (x.bic, party)) partyBics
+      pure $ TextMap.union result partyBicMapping)
+    TextMap.empty
+    knownParties
+  let entries = mapOptional
```

Comment:
I would just "return" this (without let binding and then returning)

Reply:
done